### PR TITLE
Feat(2)/seeding and fixtures

### PR DESCRIPTION
**Pull request by jens-smeds-zupit**
An initial version of the tool suggested in issue #2.

- The library exposes interfaces `ISeed` and `IFixture` as well as an attribute for signaling that a seed/fixture depends on another seed/fixture. 
- It registers a hosted service that applies the implementations of the interfaces mentioned above.
- There is a simple store implementation that persists the task executions (an execution of a seed or a fixture).  The persistence expects a PostgreSQL database and creates a table containing execution results (success or failure+stacktrace). It also contains a timestamp of when the hosted service started to allow for grouping seeds and fixtures that have been executed as part of the same run.

### CODE REVIEW

**Comment by AleSaiani** [comment]
I really like the work done, 
I just have a slight doubt about the persistence part. 
Currently, we create a specific DbContext, which makes sense as it operates in a different schema to stay out of a domain's entities context. 
For consistency, I would have also introduced this schema change for the __EFMigrationHistory. 
Using this solution, though, we could probably use a single DbContext. 
We would always have a DbContext if we need the Seed/Fixture mechanism. 
We also have an auto-discovery mechanism for classes to be mapped as persistent or with the classic system based on EF Core's IEntityTypeConfiguration. 
And this permit to have only "one" migration in the system.
So, to summarize, I would propose to:

- keep a single DbContext
- implement IEntityTypeConfiguration for TaskExecution to be discovered and created by "application" DbContext
- move the migration table as well into another schema (we could use a common "devops" schema)

What do you think?

**Comment by jens-smeds-zupit** [comment]
> Maybe we also can introduce a feature (an attribute?) to specificy a different schema for entities (there is a Table Attribute of EF Core that maybe can fit in this idea, ex. [Table("StudentMaster", Schema="Admin")])
> 
> What do you think?
> 
> p.s. another idea I can suggest to explore is if we can use an "Empty" Migration as container for our Seed/Fixture code to use the MigrationHistory to keep track of execution

We did it using a separate DbContext due to the following 3 reasons:

1. Keeping transactions clearly separate.
2. Avoid being very opinionated regarding the persistence mechanism of the consumer.
3. Using a DbContext proved faster than going with some more light weight solution (like Dapper).

Considering that the Pragmatic Design as such is built around EF, point 2 is not so important. 
We can have a look at your proposed solution (and also se how to deal with point 1).
